File: Week_1/Day_4/2.py
#2.Reshape the Matrix
class Solution:
    def matrixReshape(self, mat: list[list[int]], r: int, c: int) -> list[list[int]]:
        m=len(mat)
        n=len(mat[0])
        if m*n != r*c:
            return mat
        new=[[0]*c for _ in range(r)]
        count=0
        for row in range(m):
            for col in range(n):
                new[count//c][count%c]=mat[row][col]
                count+=1
        return new

File: Week_2/Day_1/2.cpp
#include <iostream>
#include <string>
#include <vector>

class Solution {
public:
    int firstUniqChar(std::string s) {
        std::vector<int> count(26, 0);
        for (char c : s) {
            count[c - 'a']++;
        }
        for (int i = 0; i < s.length(); i++) {
            if (count[s[i] - 'a'] == 1) {
                return i; 
            }
        }
        return -1; 
    }
};